vehicle/polestar/identity.go

```diff
@@ -137,3 +147,41 @@ func (v *Identity) RefreshToken(token *oauth2.Token) (*oauth2.Token, error) {
 
 	return v.login()
 }
+
+func (v *Identity) confirmConsentAndGetCode(uri, resume string, param request.InterceptResult) (string, error) {
+
+	// Step 1: Extract the user ID (UID) from the redirect parameters
+	var uid string
+	if uid, err := param(); err != nil || uid == "" {
```

Comment:
In my tests, there is no UID in the params, still trying to figure out why.

Reply:
We are never reading a "uid" so it also cannot be extracted. I guess there is still something missing to make that work

Reply:
added

Reply:
Does that actually work for you @rostbeule ?

For me it never retrieves the `uid`. Unfortunately, I am not really good with Go and do not fully understand how the this `InterceptRedirect` works.

There is a rather simple Python script that works perfectly fine for me:
https://github.com/CONSULitAS/Polestar_2_MQTT_Docker/blob/450503db2d1d5fce857326214b5f7709c3dea43e/Polestar_2_MQTT.py#L87

Here you can see that the `uid` is extracted from the response header after making the post request to `https://polestarid.eu.polestar.com/as/{path_token}/resume/as/authorization.ping`

I am still trying to figure out the difference to this working example but again struggle with Go 😉

Reply:
I can reassure you there. These are actually my first experiences with Go. Honestly, I tested it in C# and tried to transfer it to Go. But I was upfront about that 😅

Will have a look into that